akka-docs/src/test/java/jdocs/sharding/ClusterShardingTest.java

```diff
@@ -85,12 +85,12 @@ public String shardId(Object message) {
     //#counter-start
     Option<String> roleOption = Option.none();
     ClusterShardingSettings settings = ClusterShardingSettings.create(system);
-    ActorRef startedCounterRegion = ClusterSharding.get(system).start("Counter",
-      Props.create(Counter.class), settings, messageExtractor);
+    ActorRef startedCounterRegion = ClusterSharding.get(system).start(Counter.ShardingTypeName,
+      Counter.propsFactory, settings, messageExtractor);
```

Comment:
I think it's easer to read (understand) the docs by putting the function inline here
```
entityId -> Counter.props(entityId)
```

Reply:
Actually, what about `Counter::props` for this?
(Should've seen that from the start, guess it's been a while since I wrote actual code in Java...)

Reply:
that should work, but for the same reason I argued about using explicit entityId parameter in the scala examples we should do that here also

Reply:
Alright, in that case this build should cover it (once it's done building)